Source/Charts/Data/Implementations/Standard/BarChartDataEntry.swift

```diff
@@ -62,32 +66,27 @@ open class BarChartDataEntry: ChartDataEntry
         calcPosNegSum()
         calcRanges()
     }
-        
+
     /// Constructor for stacked bar entries. One data object for whole stack
-    @objc public init(x: Double, yValues: [Double], data: AnyObject?)
+    @objc public convenience init(x: Double, yValues: [Double], icon: NSUIImage?)
     {
-        super.init(x: x, y: BarChartDataEntry.calcSum(values: yValues), data: data)
-        self._yVals = yValues
-        calcPosNegSum()
-        calcRanges()
+        self.init(x: x, yValues: yValues)
+        self.icon = icon
     }
-    
+
     /// Constructor for stacked bar entries. One data object for whole stack
-    @objc public init(x: Double, yValues: [Double], icon: NSUIImage?, data: AnyObject?)
+    @objc public convenience init(x: Double, yValues: [Double], data: AnyObject?)
     {
-        super.init(x: x, y: BarChartDataEntry.calcSum(values: yValues), icon: icon, data: data)
-        self._yVals = yValues
-        calcPosNegSum()
-        calcRanges()
+        self.init(x: x, yValues: yValues)
+        self.data = data
     }
-    
+
     /// Constructor for stacked bar entries. One data object for whole stack
-    @objc public init(x: Double, yValues: [Double], icon: NSUIImage?)
+    @objc public convenience init(x: Double, yValues: [Double], icon: NSUIImage?, data: AnyObject?)
     {
-        super.init(x: x, y: BarChartDataEntry.calcSum(values: yValues), icon: icon)
-        self._yVals = yValues
-        calcPosNegSum()
-        calcRanges()
+        self.init(x: x, yValues: yValues)
```

Comment:
time is up for me today, so I cannot go deeper I suppose
@jjatie I want to be careful here, have you checked those function calls like `calcPosNegSum` and `calcRanges` that do not use `data` or other parameters?

The new change here seems changed the order that calls other calculation functions. e.g. in old IMP it will start calculate after all parameters are set, but with the new init(), the order is no longer there.

I will check back tomorrow

Reply:
Are you saying you want to assign `icon` and `data` before calling `self.init(x: x, yValues: yValues)`?
I'm okay with that.

Reply:
On second thought, it would be more efficient that way so I will change it.

Reply:
what I'm concerned is if we change order of the setter calls before or after the calculation functions, it might trigger weird or not easy found issues. so wanted to be careful about it. if we sure the properties are not used in the calculation, I'm fine with the new order. but it might take longer to have a review :)

Reply:
I will correct the order tonight

Reply:
There shouldn't be any affect here